app/models/data.py

```diff
@@ -264,9 +292,32 @@ def __init__(self, **kwargs):
 
         # "date" is expected to be a date string, no times or timezones
         if 'date' in kwargs:
-            kwargs['date'] = self.parse_str_to_date(kwargs['date'])
+            kwargs['date'] = CoreData.parse_str_to_date(kwargs['date'])
         else:
             kwargs['date'] = date.today()
+        return kwargs
+
+    def copy_with_updates(self, **kwargs):
```

Comment:
sorry if this is a silly question - why not just make a new `kwargs` dictionary and pass that to the `CoreData` constructor, instead of setting these values directly in this method? What would that break?

Reply:
err, this becomes annoying because of SQLAlchemy and its special classes. I can look into it later, I haven't used SQLAlchemy before, but to do this type of copy and preserving the constructor I could expunge object to have a "clean version" not connected to a session that later I could add again as a new row

Reply:
Since we have `to_dict()` methods on these objects, would it work to extract that, do updates within that dict and then make a new `CoreData` object from that updated dict as kwargs? Sorry if sending you on a wild goose chase, let me know if this gets annoying. Hoping to avoid a lot of SQLAlchemy magic in the API code in general. Is what you're trying to do ultimately to create a new CoreData object identical in field values to this one, with the exception of the overriding kwargs as passed in here?

Reply:
Yeah, this seems to work. Donno whether I tried it and something didnt work in the past... can't recall